File: 0520-detect-capital/0520-detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int n = word.size();
        int upperCount = 0;

        for(auto c: word){
            if(isupper(c)){
                upperCount++;
            }
        }
        if(upperCount == n){
            return true;
        }
        if(upperCount == 0){
            return true;
        }
        if(upperCount == 1 && isupper(word[0])){
            return true;
        }
        else{
            return false;
        }
    }
};

File: 0520-detect-capital/0520-detect-capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        n = len(word)
        upperCount = 0
        for c in word:
            if c.isupper():
                upperCount += 1

        if upperCount == n:
            return True
        if upperCount == 0:
            return True
        if upperCount == 1 and word[0].isupper():
            return True
        else:
            return False


        